### PR TITLE
Handle azure oci manifest

### PR DESCRIPTION
**Pull request by adarsh-zededa**
Description of this issue (by @deitch):
AzureContainerRregistry actually sanely implements the OCI distribution spec, whereas Docker Hub does not. We ended up relying on the DHub implementation.

when you want to pull a blob, you use endpoint /v2/<name>/blobs/<digest>
when you want to pull a manifest, you use endpoint /v2/<name>/manifests/<reference>
What do you do when you have a sha256 but do not know if it is a manifest-type (manifest or index) or something else (layer or config)? The sane thing to do is have it always available via /blob, since a blob is a blob is a blob.

Docker Hub does not do that, it returns an error if what you are requesting is a manifest; Azure returns the content.

Signed-off-by: adarsh-zededa <adarsh@zededa.com>

### CODE REVIEW

**Review by rvs — APPROVED** [review]
Please fix yetus complaints -- LGTM otherwise

**Comment by deitch** [comment]
It also is getting build errors.

File: libs/zedUpload/ociutil/oci.go [hunk 1]
```diff
@@ -127,15 +126,11 @@ func PullBlob(registry, repo, hash, localFile, username, apiKey string, maxsize
...
-		r, contentType, err = ociGetManifest(ref, opts)
-		if err != nil {
-			return 0, "", err
-		}
-	} else {
-		// we had a hash, so get the actual layer, but fall back to manifest
+	// check if we have a manifest
+	r, contentType, err = ociGetManifest(ref, opts)
+	if err != nil {
+		logrus.Infof("PullBlob(%s): unable to fetch manifest. %s", image, err.Error())
```
Comment (deitch) [inline libs/zedUpload/ociutil/oci.go:132]: With the change, this log line is confusing, and will look like an error. Might want to add a line before like, "trying to get <ref> as manifest", and then this line can be, "unable to fetch <ref> as manifest, trying as blob", etc.

**Review by deitch — CHANGES_REQUESTED** [review]
The logic looks pretty good, and hats off, it looks a lot cleaner than before! The only changes I would request are in the logging, which might look confusing now.

**Review by deitch — APPROVED** [review]

**Review by rvs — APPROVED** [review]
LGTM, but seems like Eden is failing -- I've re-triggered it -- but if it fails again you'll have to look into this @adarsh-zededa

**Pull request merged**